spec/formalism/form/coercion_spec.rb

```diff
@@ -0,0 +1,189 @@
+# frozen_string_literal: true
+
+## This is unit test, created to simplify testing of different coercions
+## You can find integration test in spec/formalism/form/fields_spec.rb
+## (Formalism::Form -> .field -> coersion)
+##
+## In tests `eql` used instead of `eq` to determine result type
+describe Formalism.const_get(:Coercion) do
+	let(:value) {}
+	let(:type) {}
+
+	subject(:coercion_result) { described_class.new(value, type: type).result }
+
+	shared_examples 'it parses nil' do |expected_result = nil|
+		let(:value) { nil }
+		it { is_expected.to eql(expected_result) }
+	end
+
+	shared_examples 'it parses empty string' do |expected_result = nil|
```

Comment:
I think, `expected_result` should be required argument.

Reply:
> I think, expected_result should be required argument.

`nil`  is used quiet often, and i sure this semantic should be common for most types will be added later

Reply:
OK, I didn't notice arguments used quietly often.